engine/runner/src/main/java/org/enso/runner/Main.java

```diff
@@ -1459,6 +1466,80 @@ final CommandLine preprocessArguments(String... args) {
     }
   }
 
+  /**
+   * This method has to be called as early as possible. It attempts to find the project root
+   * directory of the given file, and if the project root is found, it uses native code to change
+   * the working directory to the project root. In order for the JVM's {@code java.io} to reflect
+   * the working directory change, this methods must be called before any class from {@code java.io}
+   * is accessed.
+   *
+   * <p>Note that invoking native code is the only reliable way to change the working directory in
+   * the current process.
+   *
+   * <p>For detailed explanation see this <a
+   * href="https://github.com/enso-org/enso/pull/12618#issuecomment-2778451448">GH comment</a>.
+   *
+   * @param fileToRun the file to run, value of the {@code --run} option.
+   */
+  private void maybeChangeWorkingDirToProjectRoot(String fileToRun) {
+    assert fileToRun != null;
+    if (!ImageInfo.inImageRuntimeCode()) {
+      logger.warn("Changing working directory only supported in native image");
```

Comment:
- I assume there is going to be the "other warning" printed later
- No need to warn, if the current directory is correct
- Change to `logger.trace`, maybe?